Laravel-Backpack/CRUD
# [Bug] Fix column attribute for filters

## WHY

### BEFORE - What was wrong? What was happening before this PR?

[#4293](https://github.com/Laravel-Backpack/CRUD/issues/4293) explains it very good, but in short, in some cases the filtering function in **List Operation** was not working due to the columns lacking the `attribute` key.

### AFTER - What is happening after this PR?

Filtering should work as expected.


## HOW

### How did you achieve that, in technical terms?

I based the logic on `makeSureFieldHasAttribute()` used for fields to add the attribute when needed.



### Is it a breaking change?

No.


### How can we test the before & after?

I used Products and its relationship with categories. I set the category column in `ProductCrudController` like this:

`CRUD::column('category_id')->type('select');` Yes, I had to specify the `select` type to reproduce the error.
After that, search anything in filter. That should trigger an error.

Checkout branch `fix-column-for-attribute-for-filters` and perform another search. Now it should work.
